projects/iio_adpd1080/src/main.c

```diff
@@ -0,0 +1,254 @@
+#include "error.h"
+#include "util.h"
+#include "iio.h"
+#include "irq.h"
+#include "irq_extra.h"
+#include "uart.h"
+#include "uart_extra.h"
+#include "iio_adpd188.h"
+#include "i2c.h"
+#include "iio_app.h"
+
+#define MAX_SIZE_BASE_ADDR		1024
+
+static uint8_t in_buff[MAX_SIZE_BASE_ADDR];
+
+#define ADC_DDR_BASEADDR	((uint32_t)in_buff)
+
+/***************************************************************************//**
+ * @brief main
+*******************************************************************************/
+int main(void)
+{
+	int32_t status;
+	uint16_t reg_data;
+
+	status = platform_init();
+	if (IS_ERR_VALUE(status))
+		return status;
+
+	struct adpd188_iio_desc *adpd1080_iio_device;
+	struct adpd188_iio_init_param adpd1080_iio_inital = {
+		.drv_init_param.device = APDP1080,
+		.drv_init_param.phy_opt = ADPD188_I2C,
+		.drv_init_param.phy_init.i2c_phy =
+		{
+			.slave_address = 0x64,
+			.max_speed_hz = 400000,
+			.platform_ops = NULL,
+			.extra = NULL
+		},
+		.drv_init_param.gpio0_init =
+		{
+			.number = 0,
+			.platform_ops = NULL,
+			.extra = 0
+		},
+		.drv_init_param.gpio1_init =
+		{
+			.number = 0,
+			.platform_ops = NULL,
+			.extra = 0
+		}
+	};
+	struct iio_data_buffer iio_adpd1080_read_buff = {
+		.buff = ADC_DDR_BASEADDR,
+		.size = MAX_SIZE_BASE_ADDR,
+	};
+
+	status = adpd188_iio_init(&adpd1080_iio_device, &adpd1080_iio_inital);
+	if (status < 0)
+		return status;
+
+	status = adpd188_adc_fsample_set(adpd1080_iio_device->drv_dev, 16.0);
+	if(status != SUCCESS)
+		return FAILURE;
+
+	status = adpd188_reg_read(adpd1080_iio_device->drv_dev, ADPD188_REG_SLOT_EN,
+				  &reg_data);
+	if(status != SUCCESS)
+		return FAILURE;
+	reg_data |= ADPD188_SLOT_EN_RDOUT_MODE_MASK |
+		    ADPD188_SLOT_EN_FIFO_OVRN_PREVENT_MASK;
+	status = adpd188_reg_write(adpd1080_iio_device->drv_dev, ADPD188_REG_SLOT_EN,
+				   reg_data);
+	if(status != SUCCESS)
+		return FAILURE;
+
+	/* Slot A chop mode is inverted, non-inverted, non-inverted, inverted */
+	status = adpd188_reg_read(adpd1080_iio_device->drv_dev, ADPD188_REG_INT_SEQ_A,
+				  &reg_data);
+	if(status != SUCCESS)
+		return FAILURE;
+	reg_data |= 0x9 & ADPD188_INT_SEQ_A_INTEG_ORDER_A_MASK;
+	status = adpd188_reg_write(adpd1080_iio_device->drv_dev, ADPD188_REG_INT_SEQ_A,
+				   reg_data);
+	if(status != SUCCESS)
+		return FAILURE;
+	/* Slot B chop mode is inverted, non-inverted, non-inverted, inverted */
+	status = adpd188_reg_read(adpd1080_iio_device->drv_dev, ADPD188_REG_INT_SEQ_B,
+				  &reg_data);
+	if(status != SUCCESS)
+		return FAILURE;
+	reg_data |= 0x9 & ADPD188_INT_SEQ_B_INTEG_ORDER_B_MASK;
+	status = adpd188_reg_write(adpd1080_iio_device->drv_dev, ADPD188_REG_INT_SEQ_B,
+				   reg_data);
+	if(status != SUCCESS)
+		return FAILURE;
+
+	/* Set blue LED 1 power */
+	status = adpd188_reg_read(adpd1080_iio_device->drv_dev,
+				  ADPD188_REG_ILED1_COARSE, &reg_data);
+	if(status != SUCCESS)
+		return FAILURE;
+	reg_data &= ~ADPD188_ILED1_COARSE_ILED1_COARSE_MASK;
```

Comment:
Are this some generic configurations?  Isn't any function in the driver to do this instead of using direct register access?
It is easy for the user to do something wrong working directly with the registers.

Reply:
Those configurations are specific to the PMOD application. Should we add some dedicated functions?

Reply:
I would say it's your decision, it was just a suggestion. :)